sdk/identity/src/client_credentials_flow/mod.rs

```diff
@@ -64,29 +61,19 @@ pub async fn perform(
     let url = url::Url::parse(&format!(
         "https://login.microsoftonline.com/{}/oauth2/v2.0/token",
         tenant_id
-    ))
-    .map_err(|error| AzureError::GenericErrorWithText(error.to_string()))?;
+    ))?;
 
     client
         .post(url)
         .header("ContentType", "Application / WwwFormUrlEncoded")
         .body(encoded)
         .send()
-        .await
-        .map_err(|e| AzureError::GenericErrorWithText(e.to_string()))?
+        .await?
         .text()
-        .map_err(|e| AzureError::GenericErrorWithText(e.to_string()))
         .await
-        .and_then(|s| {
-            serde_json::from_str::<LoginResponse>(&s).map_err(|e| {
-                serde_json::from_str::<errors::ErrorResponse>(&s)
-                    .map(|er| AzureError::GenericErrorWithText(er.to_string()))
-                    .unwrap_or_else(|_| {
-                        AzureError::GenericErrorWithText(format!(
-                            "Failed to parse Azure response: {}",
-                            e.to_string()
-                        ))
-                    })
-            })
-        })
+        .map(|s| -> Result<LoginResponse, Error> {
+            Ok(serde_json::from_str::<LoginResponse>(&s)?)
+        })?
+    // TODO The HTTP status code should be checked to deserialize an error response.
```

Comment:
Can we create an issue for this?

Reply:
https://github.com/Azure/azure-sdk-for-rust/issues/270